stl/inc/memory

```diff
@@ -1596,18 +1887,307 @@ private:
+        _Maybe_rebind_alloc_t<_Alloc, remove_cv_t<remove_all_extents_t<_Ty>>> _Alty(this->_Get_val());
+        _Uninitialized_fill_multidimensional_n_al(_Get_ptr(), _Size, _Val, _Alty);
+    }
+
+    auto _Get_ptr() {
+        return reinterpret_cast<remove_extent_t<_Ty>*>(
+            reinterpret_cast<char*>(this) + _Div_ceil(sizeof(*this), alignof(_Ty)) * alignof(_Ty));
+    }
+
+private:
+    size_t _Size;
+
+    ~_Ref_count_unbounded_array_alloc() = default; // nothing to do; _Storage._Value already destroyed by _Destroy()
+
+    virtual void _Destroy() noexcept override { // destroy managed resource
+        if constexpr (!conjunction_v<is_trivially_destructible<remove_extent_t<_Ty>>,
+                          _Uses_default_destroy<_Alloc, _Ty*>>) {
+            _Maybe_rebind_alloc_t<_Alloc, remove_all_extents_t<_Ty>> _Alty(this->_Get_val());
+            _Reverse_destroy_multidimensional_n_al(_Get_ptr(), _Size, _Alty);
+        }
+    }
+
+    virtual void _Delete_this() noexcept override { // destroy self
+        constexpr size_t _Align = (_STD max)(alignof(_Ty), alignof(_Ref_count_unbounded_array_alloc));
+        using _Storage          = aligned_storage_t<_Align, _Align>;
+        _Rebind_alloc_t<_Alloc, _Storage> _Al(this->_Get_val());
+        this->~_Ref_count_unbounded_array_alloc();
+        allocator_traits<_Rebind_alloc_t<_Alloc, _Storage>>::deallocate(_Al, reinterpret_cast<_Storage*>(this),
+            _Div_ceil(_Size * sizeof(remove_extent_t<_Ty>), sizeof(_Storage))
+                + _Div_ceil(sizeof(*this), sizeof(_Storage)));
+    }
+};
+
+// CLASS TEMPLATE _Ref_count_bounded_array_alloc
+template <class _Ty, class _Alloc>
+class __declspec(empty_bases) _Ref_count_bounded_array_alloc : public _Ebco_base<_Alloc>, public _Ref_count_base {
+    // handle reference counting for object in control block, allocator
+public:
+    explicit _Ref_count_bounded_array_alloc(const _Alloc& _Al_arg) : _Ebco_base<_Alloc>(_Al_arg), _Ref_count_base() {
+        _Maybe_rebind_alloc_t<_Alloc, remove_cv_t<remove_all_extents_t<_Ty>>> _Alty(this->_Get_val());
+        _Uninitialized_value_construct_multidimensional_n_al(_Storage._Value, extent_v<_Ty>, _Alty);
+    }
+
+    explicit _Ref_count_bounded_array_alloc(const _Alloc& _Al_arg, const remove_extent_t<_Ty>& _Val)
+        : _Ebco_base<_Alloc>(_Al_arg), _Ref_count_base() {
+        _Maybe_rebind_alloc_t<_Alloc, remove_cv_t<remove_all_extents_t<_Ty>>> _Alty(this->_Get_val());
+        _Uninitialized_fill_multidimensional_n_al(_Storage._Value, extent_v<_Ty>, _Val, _Alty);
+    }
+
+    union {
+        _Wrap<_Ty> _Storage;
+    };
+
+private:
+    ~_Ref_count_bounded_array_alloc() {
+        // nothing to do; _Storage._Value already destroyed by _Destroy()
+    }
+
+    virtual void _Destroy() noexcept override { // destroy managed resource
+        if constexpr (!conjunction_v<is_trivially_destructible<_Ty>, _Uses_default_destroy<_Alloc, _Ty*>>) {

```

Comment:
We should double-check both occurrences of `_Uses_default_destroy` - I am unsure whether the template arguments are correct.

Reply:
They should be correct now

Reply:
I think there's still an issue when `_Maybe_rebind_alloc_t` emits a reference, because `_Uses_default_destroy` asks `_Is_default_allocator` and that recognizes `allocator<_Ty>` but not references.

Reply:
Fixed